src/client/ClientService.ts

```diff
@@ -16,34 +16,32 @@ import * as errors from './errors';
 import config from '../config';
 import * as networkUtils from '../network/utils';
 
-interface ClientService extends CreateDestroyStartStop {}
-@CreateDestroyStartStop(
-  new errors.ErrorClientServiceRunning(),
-  new errors.ErrorClientServiceDestroyed(),
-  {
-    eventStart: events.EventClientServiceStart,
-    eventStarted: events.EventClientServiceStarted,
-    eventStop: events.EventClientServiceStop,
-    eventStopped: events.EventClientServiceStopped,
-    eventDestroy: events.EventClientServiceDestroy,
-    eventDestroyed: events.EventClientServiceDestroyed,
-  },
-)
+interface ClientService extends StartStop {}
+@StartStop({
+  eventStart: events.EventClientServiceStart,
+  eventStarted: events.EventClientServiceStarted,
+  eventStop: events.EventClientServiceStop,
+  eventStopped: events.EventClientServiceStopped,
+})
 class ClientService {
-  public static async createClientService({
-    manifest,
-    tlsConfig,
-    options: {
-      middlewareFactory,
-      host = config.defaultsUser.clientServiceHost,
-      port = config.defaultsUser.clientServicePort,
-      keepAliveTimeoutTime = config.defaultsSystem.clientKeepAliveTimeoutTime,
-      keepAliveIntervalTime = config.defaultsSystem.clientKeepAliveIntervalTime,
-      rpcCallTimeoutTime = config.defaultsSystem.rpcCallTimeoutTime,
-      rpcParserBufferSize = config.defaultsSystem.rpcParserBufferSize,
-    },
-    logger = new Logger(this.name),
-  }: {
+  protected rpcServer: RPCServer;
+  protected webSocketServer: WebSocketServer;
+  protected logger: Logger;
+
+  public constructor({
+                       manifest,
+                       tlsConfig,
+                       options: {
+                         middlewareFactory,
+                         host = config.defaultsUser.clientServiceHost,
+                         port = config.defaultsUser.clientServicePort,
+                         keepAliveTimeoutTime = config.defaultsSystem.clientKeepAliveTimeoutTime,
+                         keepAliveIntervalTime = config.defaultsSystem.clientKeepAliveIntervalTime,
+                         rpcCallTimeoutTime = config.defaultsSystem.rpcCallTimeoutTime,
+                         rpcParserBufferSize = config.defaultsSystem.rpcParserBufferSize,
+                       },
+                       logger = new Logger(),
+                     }: {
```

Comment:
New logger needs a name.

Reply:
Gotcha

Reply:
Its not lintfixed yet so stuff might look a lil jank, I'll lint it before merge